service/src/main/java/tds/session/services/impl/SessionServiceImpl.java

```diff
@@ -31,38 +40,55 @@ public SessionServiceImpl(SessionRepository sessionRepository,
     }
 
     @Override
-    public Optional<PauseSessionResponse> pause(final UUID sessionId, final String newStatus) {
-        final Optional<Session> sessionOptional = sessionRepository.findSessionById(sessionId);
-        if (!sessionOptional.isPresent()) {
-            return Optional.empty();
-        }
-
-        Session session = sessionOptional.get();
-        PauseSessionResponse pauseSessionResponse;
+    public Response<PauseSessionResponse> pause(final UUID sessionId, PauseSessionRequest request) {
+        final Session session = sessionRepository.findSessionById(sessionId)
+            .orElseThrow(() -> new NotFoundException(String.format("Could not find session for session id %s", sessionId)));
 
-        // if session is not open, it does not need to be paused, so return.
-        // TODO:  This condition might be over-simplified
-        if (!session.isOpen()) {
-            // TODO:  Collect all Exams associated w/the Session
-            //pauseSessionResponse.setExamIds(examIds);
-            pauseSessionResponse = new PauseSessionResponse(session);
-            return Optional.of(pauseSessionResponse);
+        Optional<ValidationError> maybeValidationError = verifySessionCanBePaused(session, request);
+        if (maybeValidationError.isPresent()) {
+            return new Response<PauseSessionResponse>(maybeValidationError.get());
         }
 
-        // TODO:  Call pause exam for each Exam in session and set affected examIds list on response object
-        // pauseSessionResponse.setExamIds();
+        examService.pauseAllExamsInSession(sessionId);
+        sessionRepository.pause(sessionId, "closed");
 
-        // Pause the Session
-        sessionRepository.pause(sessionId, newStatus);
+        Session updatedSession = sessionRepository.findSessionById(sessionId)
```

Comment:
I think this should be a 500 Exception.  If we get here and can't find the session we just paused something is seriously wrong.